src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetVersionService.cs

```diff
@@ -100,6 +123,115 @@ private static DataSetVersionSummaryViewModel MapDraftSummaryVersion(DataSetVers
+    }
+
+    private static Either<ActionResult, Unit> CheckCanUpdateVersion(
+        DataSetVersion dataSetVersion,
+        DataSetVersionUpdateRequest updateRequest)
+    {
+        if (!dataSetVersion.CanBeUpdated)
+        {
+            return ValidationUtils.ValidationResult(new ErrorViewModel
+            {
+                Code = ValidationMessages.DataSetVersionCannotBeUpdated.Code,
+                Message = ValidationMessages.DataSetVersionCannotBeUpdated.Message,
+                Detail = new InvalidErrorDetail<Guid>(dataSetVersion.Id),
+                Path = nameof(DataSetVersionUpdateRequest.DataSetVersionId).ToLowerFirst()
+            });
+        }
+
+        if (updateRequest.Notes is not null && dataSetVersion.IsFirstVersion)
+        {
+            return ValidationUtils.ValidationResult(new ErrorViewModel
+            {
+                Code = ValidationMessages.DataSetVersionCannotHaveChangelogNotes.Code,
+                Message = ValidationMessages.DataSetVersionCannotHaveChangelogNotes.Message,
+                Detail = new InvalidErrorDetail<Guid>(dataSetVersion.Id),
+                Path = nameof(DataSetVersionUpdateRequest.Notes).ToLowerFirst()
+            });
+        }
+
+        return Unit.Instance;
+    }
+
+    private async Task<Either<ActionResult, DataSetVersion>> UpdateVersion(
+        DataSetVersion dataSetVersion,
+        DataSetVersionUpdateRequest updateRequest,
+        CancellationToken cancellationToken)
+    {
+        if (updateRequest.Notes is not null)
+        {
+            dataSetVersion.Notes = updateRequest.Notes;
+        }
+
+        publicDataDbContext.DataSetVersions.Update(dataSetVersion);
+
+        await publicDataDbContext.SaveChangesAsync(cancellationToken);
+
+        return dataSetVersion;
+    }
+
+    private async Task<DataSetDraftVersionViewModel> MapVersion(DataSetVersion dataSetVersion)
+    {
+        var releaseFile = await GetReleaseFile(dataSetVersion);
+
+        return new DataSetDraftVersionViewModel
+        {
+            Id = dataSetVersion.Id,
+            Version = dataSetVersion.Version,
+            Status = dataSetVersion.Status,
+            Type = dataSetVersion.VersionType,
+            File = MapVersionFile(releaseFile),
+            ReleaseVersion = MapReleaseVersion(releaseFile.ReleaseVersion),
+            TotalResults = dataSetVersion.TotalResults,
+            Notes = dataSetVersion.Notes,
+            GeographicLevels = dataSetVersion.MetaSummary?.GeographicLevels
+                .Select(l => l.GetEnumLabel())
+                .ToList() ?? null,
+            TimePeriods = dataSetVersion.MetaSummary?.TimePeriodRange is not null
+                ? TimePeriodRangeViewModel.Create(dataSetVersion.MetaSummary.TimePeriodRange)
+                : null,
+            Filters = dataSetVersion.MetaSummary?.Filters ?? null,
+            Indicators = dataSetVersion.MetaSummary?.Indicators ?? null
+        };
+    }
```

Comment:
Just wondering if we should call this `MapDraftVersion` like we do in `DataSetService`, although that could be because that also has a `MapLiveVersion`.

There's kind of a loose connection implied her,e that because the data set version `CanBeUpdated` property is false, that it must be a draft version (or one of the other draft statuses), and so we are mapping to `DataSetDraftVersionViewModel` rather than one of the other view models like `DataSetLiveVersionViewModel`.

I'm not sure whether there's anything we can do to improve on this? Perhaps a check at the beginning of this method which would throw an exception if the status is not one of the draft statuses?

Reply:
I agree `MapDraftVersion` would be better, I've renamed it :) . We no longer cover **all** of the draft statuses when updating now, but `MapDraftVersion` is still correctly named even if it will only be called with a subset of all the possible draft statuses.

In terms of adding a check, I'm not sure if this is necessary as we will be duplicating the check on line 141 in `CheckCanUpdateVersion`. I guess when looking at this method in isolation, it can be confusing though.

**EDIT:** Thinking out loud though - If someone comes along in the future and tries to re-use this method for mapping draft versions, then it should probably throw an error if it isn't a draft status. But in this particular scenario, we don't want to cover **all** draft statuses, only the ones that allow a version to be updated. So....... maybe the exception should be thrown if it isn't `Failed` or `Mapping`? And the method should more accurately be called `MapUpdateableVersion`. But I don't like that either!


What do you think?

Reply:
I can't think of a good solution here that avoids duplicating the check already made on L141 and makes it reusable by any future work that needs to map draft versions.

It would also be a bit weird to call that check again here since it's named `CanBeUpdated` whereas you'd expect to see a check like `IsDraft`.

I think that given we don't have anything else using this right now and the `CanBeUpdated` check is already being made outside of this, that we leave it for now with just the rename to `MapDraftVersion`. 🙂

Reply:
I've gone ahead with your suggestion and just done the rename for now :)